pm/process_pm.f90

```diff
@@ -0,0 +1,143 @@
+program pmbufr
+
+ implicit none
+
+ integer, parameter :: mxmn=35, mxlv=255, maxcnt=5000, pm_limit=-15
+ integer :: ireadmg,ireadsb,idate,nmsg,ntb,nsubset,nlvl
+ integer :: i,j,k,NARG,iret_hdr,iret_ob,interval
+ integer :: dhr,iret_hdr1,iret_ob1,cnt
+ integer :: unit_table,unit_out,unit_site,unit_var,nt,pm_len
+
+ character(80):: hdstr='SID XOB YOB DHR TYP T29 SQN PROCN RPT CAT TYPO TSIG'
+ character(80):: obstr='TPHR QCIND COPOPM'
+ character(8) :: subset,c_sid
+ character(10):: analysis_time
+ character(20):: str,infile,outfile
+ character(20),allocatable,dimension(:)::sta_id, lat_str, lon_str, elv_str, pm_aqi_str, pm_measured_str, pm_str
+
+ real*8,parameter:: bmiss=10e10_8
+ real(8) :: hdr(mxmn),obs(mxmn,mxlv)
+ real(8) :: rstation_id, lat, lon, elv, pm
+
+ logical :: ifexist
+
+ namelist/setup/analysis_time,infile,outfile,cnt
+
+ ! only c_sid*8 is encoded to bufr file since SID is 64 bit in bufrtable
+ equivalence(rstation_id,c_sid)
+
+! read namelist
+ inquire(file='namelist.pm', EXIST=ifexist )
+ if(ifexist) then
+   open(15, file='namelist.pm')
+     read(15,setup)
+   close(15)
+   write(*,'(a45,a20,3x,2a20,i8)') 'analysis_time, infile, outfile, cnt: ' ,analysis_time, infile, outfile, cnt    
+   write(*,*)
+ else
+   write(*,*) "Usage: analysis_time, infile, outfile, pm_count"
+ stop 123
+
+ endif
+
+ if (cnt==0) then
+   write(*,*) 'No observation available, exit'
+   stop
+ end if
+
+! open and read pm2.5 ascii file
+ unit_var=11
+ allocate(sta_id(cnt))
+ allocate(lat_str(cnt))
+ allocate(lon_str(cnt))
+ allocate(elv_str(cnt))
+ allocate(pm_aqi_str(cnt))
+ allocate(pm_measured_str(cnt))
+ allocate(pm_str(cnt))
+
+ open (unit = unit_var, file = infile,status = 'old')
+ do i = 1,cnt
+   read(unit_var,'(7a16)')            sta_id(i), lat_str(i), lon_str(i), elv_str(i), pm_aqi_str(i), pm_measured_str(i), pm_str(i)
+   !write(*,'(a10,7a15)')     'obs  ', trim(adjustl(sta_id(i))), trim(adjustl(lat_str(i))), trim(adjustl(lon_str(i))),trim(adjustl(elv_str(i))), &
+   !                                   trim(adjustl(pm_aqi_str(i))), trim(adjustl(pm_measured_str(i))), trim(adjustl(pm_str(i)))
+   !write(*,'(a10,7i15)')     'len  ', len(trim(adjustl(sta_id(i)))), len(trim(adjustl(lat_str(i)))), len(trim(adjustl(lon_str(i)))), len(trim(adjustl(elv_str(i)))), &
+   !                                   len(trim(adjustl(pm_aqi_str(i)))), len(trim(adjustl(pm_measured_str(i)))), len(trim(adjustl(pm_str(i))))
+ end do
```

Comment:
remove line 61-64  if they are not needed.

Reply:
done